terraform/lib/dependabot/terraform/file_updater.rb

```diff
@@ -80,6 +86,21 @@ def update_registry_declaration(new_req, old_req, updated_content)
         end
       end
 
+      def update_lockfile_declaration(new_req, old_req, updated_content)
+        return unless lock_file?(filename)
+
+        provider_source = new_req[:source][:registry_hostname] + "/" + new_req[:source][:module_identifier]
+        declaration_regex = lockfile_declaration_regex(provider_source)
+
+        copy_dir_to_temporary_directory do
+          updated_content.sub!(declaration_regex, "")
+          
+          system "terraform providers lock #{provider_source}"
```

Comment:
Small drive by comment: we've got this util to run shell commands with escaped arguments

```suggestion
          `SharedHelpers.run_shell_command("terraform providers lock #{provider_source}")
```